src/ripple/rpc/handlers/AccountChannels.cpp

```diff
@@ -76,25 +76,29 @@ doAccountChannels(RPC::JsonContext& context)
     if (!ledger)
         return result;
 
-    std::string strIdent(params[jss::account].asString());
-    AccountID accountID;
-
-    if (auto const err = RPC::accountFromString(accountID, strIdent))
-        return err;
+    std::string const strIdent(params[jss::account].asString());
+    auto const accountID = parseBase58<AccountID>(strIdent);
+    if (!accountID)
+    {
+        rpcError(rpcACT_MALFORMED);
+    }
 
-    if (!ledger->exists(keylet::account(accountID)))
+    if (!ledger->exists(keylet::account(*accountID)))
         return rpcError(rpcACT_NOT_FOUND);
 
     std::string strDst;
     if (params.isMember(jss::destination_account))
         strDst = params[jss::destination_account].asString();
     auto hasDst = !strDst.empty();
 
-    AccountID raDstAccount;
+    std::optional<AccountID> raDstAccount{AccountID()};
```

Comment:
Can you explain this line to me and why it was changed?

Reply:
The old code calls [RPC::accountFromString()](https://github.com/XRPLF/rippled/blob/develop/src/ripple/rpc/impl/RPCHelpers.cpp#L85) with  `bStrict = false` (default value). This function attempts different ways to parse the input string (it checks whether the input is a public key, an account ID or a seed). If one of these attempts works, the function fills the content of the pointer of type `AccountID` and returns an empty Json object. Otherwise, the function returns a Json object that contains an error code and an error message.

Since we don't accept seed as a way to look up an account, I call the function [parseBase58()](https://github.com/XRPLF/rippled/blob/develop/src/ripple/protocol/impl/AccountID.cpp#L114) instead. This function has a different syntax. It returns an `std::optional<AccountID>` object that only contains an `AccountID` when the input string is actually an account ID (a valid hash).